### PR TITLE
Dont use f-strings for logging

### PR DESCRIPTION
**Pull request by Sengolda**
It is recommended in the python docs itself to not use f strings for logging.

### CODE REVIEW

**Comment by Voyz** [comment]
Hey @Sengolda thanks for your contribution to SuperLoops! I'm glad to see that you're interested in optimizing the logging 🥳😊

I've had a bit of a read about the optimisation of logging, and I didn't come to a clear conclusion as to whether this optimisation is worth introducing. Feel free to present your ideas, as I'm open to discussion on this, and correct me if I got something wrong. Here are my thoughts so far:

- I run under the assumption that the f-strings are more readable and easy to write and maintain than % format. I suppose this could be argued by some developers, but that's my perspective on it and there doesn't seem to be a clear indication from the Python docs as to which should be used.
- I cannot find any clear indication of the speed improvements this optimisation would bring. The only thing that the [Python doc](https://docs.python.org/3/howto/logging.html#optimization) states is:
    _Formatting of message arguments is deferred until it cannot be avoided._
  That is not a lot to go by. 
- Out of 23 places where we currently log, none happen on a recurrent basis - eg. in a loop. They're all one-off events, such as when a SuperLoop starts, stops, fails, is terminated, etc. These happen at such low frequency, that any optimisation to these would be unperceivable by the users.
- Seeing that the speed improvement of this optimisation would be negligible, and would reduce readability and maintainability, I'm leaning towards thinking that it is not worth introducing.

In conclusion, in my opinion the readability and maintainability benefits of using f-strings outweigh the potential speed benefits of using % format in this case.

Let me know what's your perspective on this and share any resources that could help me understand the proposed optimisation better.

Thanks! Voy
